OpenRA.Mods.Common/Traits/DockClientManager.cs

```diff
@@ -155,13 +161,39 @@ IEnumerable<IOrderTargeter> IIssueOrder.Orders
 		{
 			get
 			{
-				yield return new DockActorTargeter(
-					6,
-					Info.EnterCursor,
-					Info.EnterBlockedCursor,
-					() => requireForceMove,
-					CanQueueDockAt,
-					(target, forceEnter) => CanDockAt(target, forceEnter, true));
+				yield return new DockActorTargeter(6,
+					context =>
+					{
+						if (requireForceMove && !context.ForceEnter)
+							return CanTargetResult.Blocked(Info.EnterCursor);
+
+						if (IsTraitDisabled)
+							return CanTargetResult.Blocked(Info.EnterBlockedCursor);
+
+						var availableDockHosts = GetDockableHosts(context.Target.Actor, context.ForceEnter, context.IsQueued).ToArray();
```

Comment:
```suggestion
						var availableDockHosts = GetDockableHosts(context.Target.Actor, context.ForceEnter, context.IsQueued).ToList();
```

We should be using ToList as to generate less garbage